neuralmonkey/encoders/sentence_cnn_encoder.py

```diff
@@ -0,0 +1,278 @@
+from typing import Optional, Tuple, List, Any
+
+import tensorflow as tf
+from typeguard import check_argument_types
+
+from neuralmonkey.encoders.attentive import Attentive
+from neuralmonkey.encoders.sentence_encoder import RNNCellTuple
+from neuralmonkey.model.model_part import ModelPart, FeedDict
+from neuralmonkey.logging import log
+from neuralmonkey.nn.noisy_gru_cell import NoisyGRUCell
+from neuralmonkey.nn.ortho_gru_cell import OrthoGRUCell
+from neuralmonkey.nn.highway import highway
+from neuralmonkey.dataset import Dataset
+from neuralmonkey.vocabulary import Vocabulary
+
+
+# pylint: disable=too-many-instance-attributes
+class SentenceCNNEncoder(ModelPart, Attentive):
+    """Encoder processing a sentence using a CNN then
+    running a bidirectional RNN on the result.
+    """
+
+    # pylint: disable=too-many-arguments,too-many-locals
+    # pylint: disable=too-many-statements
+    def __init__(self,
+                 name: str,
+                 vocabulary: Vocabulary,
+                 data_id: str,
+                 embedding_size: int,
+                 segment_size: int,
+                 highway_depth: int,
+                 rnn_size: int,
+                 filters: List[Tuple[int, int]],
+                 max_input_len: Optional[int]=None,
+                 dropout_keep_prob: float=1.0,
+                 attention_type: Optional[Any]=None,
+                 attention_fertility: int=3,
+                 use_noisy_activations: bool=False,
+                 save_checkpoint: Optional[str]=None,
+                 load_checkpoint: Optional[str]=None) -> None:
+        """Create a new instance of the sentence encoder.
+
+        Based on: Jason Lee, Kyunghyun Cho, Thomas Hofmann: Fully
```

Comment:
Vychazel jsem ze sequence_cnn_encoder.py, kde to mate v konstruktoru. Myslim, ze jste ho psali vy s Jindrou :), ale klidne ti to hodim do docstringu tridy.